### PR TITLE
feat: API-28061 - update example request body generation

### PR DESCRIPTION
**Pull request by christineyost**
This PR is for API-28061. It updates example request body generation. 
First, if the MediaTypeObject `example` field is set, this field has precedence over schema examples. If this field is set, it will be used as the default request body. If it is not set, the default request body will be built from SchemaObject examples, as it is currently.
Next, currently only one request body is generated per operation. Now up to two request bodies will be generated: one with all fields containing an example and one with only required fields. If the required fields only request body will match the default request body, then only that one will be used.

### CODE REVIEW

File: src/oas-parsing/operation-example/operation-example.factory.ts [hunk 1]
```diff
@@ -9,13 +9,14 @@ export default class OperationExampleFactory {
...
       const exampleGroups = operation.exampleGroups;
+      const exampleRequestBodies = operation.exampleRequestBodies;
 
       for (const exampleGroup of exampleGroups) {
-        operationExamples.push({
-          operation,
-          exampleGroup,
-          requestBody: operation.exampleRequestBody,
-        });
+        for (const exampleRequestBody of exampleRequestBodies) {
```
Comment (mheusgit) [inline src/oas-parsing/operation-example/operation-example.factory.ts:15]: I like for loops more than most but three nested in one method calls for it to be a new function

File: src/oas-parsing/request-body/example-request-body.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { RequestBody } from 'swagger-client';
+
+class ExampleRequestBody {
```
Comment (mheusgit) [inline src/oas-parsing/request-body/example-request-body.ts:3]: Much better

File: src/suites/positive/conductors/validation-conductor.ts [hunk 3]
```diff
@@ -14,22 +15,25 @@ export default class ValidationConductor {
...
+  private operationExampleName: string;
 
   private securityValues: SecurityValues;
 
   private server: string | undefined;
 
   constructor(
     schema: OASSchema,
-    { operation, exampleGroup, requestBody }: OperationExample,
+    { operation, exampleGroup, exampleRequestBody, name }: OperationExample,
```
Comment (mheusgit) [inline src/suites/positive/conductors/validation-conductor.ts:28]: OperationExample was changed to be created with three objects but here four are returned

File: src/oas-parsing/operation-example/operation-example.factory.ts (hunk 1 above)
Comment (christineyost) [reply src/oas-parsing/operation-example/operation-example.factory.ts]: Refactored this logic

File: src/suites/positive/conductors/validation-conductor.ts (hunk 3 above)
Comment (christineyost) [reply src/suites/positive/conductors/validation-conductor.ts]: Assigned the `name` property to a variable called `operationExampleName`

**Review by mheusgit — APPROVED** [review]
Should be good to go

File: src/oas-parsing/operation-example/operation-example.factory.ts [hunk 4]
```diff
@@ -1,23 +1,49 @@
...
+    let operationExamples: OperationExample[] = [];
+
+    operations.forEach((operation) => {
+      operationExamples = [
+        ...operationExamples,
+        ...this.buildFromOperation(operation),
+      ];
+    });
+
+    return operationExamples;
```
Comment (rdalton-va) [inline src/oas-parsing/operation-example/operation-example.factory.ts:18]:
```suggestion
    return operations.flatMap((operation) => this.buildFromOperation(operation));
```

I think using [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) would help simplify this code (and below)

File: src/oas-parsing/operation-example/operation-example.factory.ts [hunk 5]
```diff
@@ -1,23 +1,49 @@
...
+    let operationExamples: OperationExample[] = [];
+
+    operation.exampleGroups.forEach((exampleGroup) => {
+      operationExamples = [
+        ...operationExamples,
+        ...this.buildFromOperationAndExampleGroup(operation, exampleGroup),
+      ];
+    });
+
+    return operationExamples;
```
Comment (rdalton-va) [inline src/oas-parsing/operation-example/operation-example.factory.ts:33]:
```suggestion
    return operation.exampleGroups.flatMap((exampleGroup) => this.buildFromOperationAndExampleGroup(operation, exampleGroup));
```

File: src/oas-parsing/operation-example/operation-example.factory.ts [hunk 6]
```diff
@@ -1,23 +1,49 @@
...
-        });
-      }
-    }
+    operation.exampleRequestBodies.forEach((exampleRequestBody) => {
+      operationExamples.push(
+        new OperationExample(operation, exampleGroup, exampleRequestBody),
+      );
+    });
 
     return operationExamples;
```
Comment (rdalton-va) [inline src/oas-parsing/operation-example/operation-example.factory.ts:48]:
```suggestion
    return operation.exampleRequestBodies.map((exampleRequestBody) => new OperationExample(operation, exampleGroup, exampleRequestBody));
```

**Review by dsteeve — APPROVED** [review]
Looks great! Thanks for the ReadMe update.

**Pull request merged**